course4/homework-string/string.hh

```diff
@@ -1,10 +1,110 @@
 #pragma once
 
+#include <cstring>
+#include <memory>
+
 namespace course {
 
 class string
 {
-	string();
+public:
+	string() : string("") {}
```

Comment:
So the default constructor will cause heap allocation? Can you try to avoid it? It's much nicer to have a cheap default ctor.